taichi/analysis/clone.cpp

```diff
@@ -136,7 +136,9 @@ class IRCloner : public IRVisitor {
     if (kernel != nullptr) {
       new_root->kernel = kernel;
     }
-    irpass::type_check(new_root.get());  // probably unnecessary
+    irpass::type_check(new_root.get(),
+                       kernel ? kernel->program.config
+                              : CompileConfig());  // TODO(#2193): remove this
```

Comment:
Will be removed in the next PR since it doesn't make sense to use `CompileConfig()`.